barretenberg/cpp/src/barretenberg/vm2/simulation/written_public_data_slots_tree_check.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "barretenberg/vm2/common/field.hpp"
+#include "barretenberg/vm2/simulation/events/written_public_data_slot_tree_check_event.hpp"
+#include "barretenberg/vm2/simulation/field_gt.hpp"
+#include "barretenberg/vm2/simulation/lib/written_slots_tree.hpp"
+#include "barretenberg/vm2/simulation/merkle_check.hpp"
+#include "barretenberg/vm2/simulation/poseidon2.hpp"
+
+namespace bb::avm2::simulation {
+
+class WrittenPublicDataSlotsTreeCheckInterface {
```

Comment:
I think this one could just have the interface of a `set`. Let's huddle I have some ideas on how to do it while still allowing passing the roots for context serialization.

(I might be too naive).